### PR TITLE
feat: verify eth signature

### PR DESCRIPTION
**Pull request by MatejVukosav**
Verify eth signature functions.

### CODE REVIEW

File: crates/identity/src/auth.rs [hunk 1]
```diff
@@ -1,13 +1,68 @@
 use libp2p::identity::Keypair;
+use web3::signing::{keccak256, recover};
 
 pub fn verify_peer_auth(keypair: &Keypair, msg: &[u8], signature: &[u8]) -> bool {
     keypair.public().verify(msg, signature)
 }
 
+pub fn verify_eth_message(account: String, message: String, signature: String) -> bool {
```
Comment (xilosada) [inline crates/identity/src/auth.rs]: technically you  are verifying the signature

File: crates/identity/src/auth.rs [hunk 2]
```diff
@@ -1,13 +1,68 @@
...
+            // Format the recovered public key as a hex string
+            let pubkey_hex = format!("{:02X}", pubkey);
+            // Compare the recovered public key with the account address in a case-insensitive manner
+            account.eq_ignore_ascii_case(&pubkey_hex)
+        }
+        Err(_) => false,
+    }
+}
+
+pub fn eth_message(message: String) -> [u8; 32] {
```
Comment (xilosada) [inline crates/identity/src/auth.rs]: str instead of String

File: crates/identity/src/auth.rs [hunk 3]
```diff
@@ -1,13 +1,68 @@
 use libp2p::identity::Keypair;
+use web3::signing::{keccak256, recover};
 
 pub fn verify_peer_auth(keypair: &Keypair, msg: &[u8], signature: &[u8]) -> bool {
     keypair.public().verify(msg, signature)
 }
 
+pub fn verify_eth_signature(account: String, message: String, signature: String) -> bool {
```
Comment (xilosada) [inline crates/identity/src/auth.rs]: str's instead of Strings

File: Cargo.toml [hunk 4]
```diff
@@ -50,6 +50,7 @@ owo-colors = "3.5.0"
 rand = "0.8.5"
 rocksdb = "0.22.0"
 sha2 = "0.10.8"
+semver = "1.0.22"
 serde = "1.0.196"
 serde_json = "1.0.113"
```
Comment (miraclx) [inline Cargo.toml:58]: let's move `sha2` past these, danke

```suggestion
semver = "1.0.22"
serde = "1.0.196"
serde_json = "1.0.113"
sha2 = "0.10.8"
```

File: crates/identity/src/auth.rs [hunk 5]
```diff
@@ -1,13 +1,68 @@
 use libp2p::identity::Keypair;
+use web3::signing::{keccak256, recover};
```
Comment (miraclx) [inline crates/identity/src/auth.rs:2]: I'd be cautious of `web3` due to all the deps it pulls in.

I'd be happy to help here. But I think we can add custom impls of these methods. They're also just delegating to other dependencies anyway.

`keccak256` - https://github.com/tomusdrw/rust-web3/blob/eccb0aa0ba846f6831a6a0149e2e318073f60f2c/src/signing.rs#L186-L193
`recover` - https://github.com/tomusdrw/rust-web3/blob/eccb0aa0ba846f6831a6a0149e2e318073f60f2c/src/signing.rs#L139-L148

Can be done in a separate PR, so as not to block this.

Comment (MatejVukosav) [reply crates/identity/src/auth.rs:2]: I have just merged this 2 minutes before your comment. xd 
Anyhow we can come back to this later again.

Comment (miraclx) [reply crates/identity/src/auth.rs:2]: 😅 yeah, that's okay

**Pull request merged**